lib/fs.js

```diff
@@ -1794,10 +1807,17 @@ function WriteStream(path, options) {
     options = {};
   else if (typeof options === 'string')
     options = { encoding: options };
-  else if (options === null || typeof options !== 'object')
-    throw new TypeError('options must be a string or an object');
-
-  options = Object.create(options);
+  else if (options === null || typeof options !== 'object') {
+    //throw new TypeError('options must be a string or an object');
+    warnedWriteStreamOptions = internalUtil.printDeprecationMessage(
+      'Passing anything but an object or a string as the second argument to ' +
+      'WriteStream is deprecated.',
+      warnedWriteStreamOptions
+    );
+    options = options || {};
+  } else {
+    options = Object.create(options);
```

Comment:
Lets say I pass a string as an argument to `ReadStream`, then the `options` will have `encoding` in its prototype. But in the WriteStream, `encoding` will be on the `options` object itself. Is this okay?